java-to-zod-gradle-plugin/src/main/java/sh/ivan/zod/GenerateZodSchemas.java

```diff
@@ -0,0 +1,321 @@
+import org.gradle.api.tasks.Optional;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.LoggerFactory;
+import sh.ivan.zod.schema.ObjectSchema;
+
+
+@Setter
+public class GenerateZodSchemas extends DefaultTask {
+
+    private static final org.slf4j.Logger log = LoggerFactory.getLogger(GenerateZodSchemas.class);
+    // Path and name of generated file
+    private File outputFile = null;
+
+    // Prefix added to generated schema names
+    private String schemaNamePrefix = "";
+
+    // Suffix added to generated schema names
+    private String schemaNameSuffix = "Schema";
+
+    // Classes to process
+    private List<String> classes = Collections.emptyList();
+
+    // Classes to process specified using glob patterns
+    private List<String> classPatterns = Collections.emptyList();
+
+    // Classes to process specified by annotations
+    private List<String> classesWithAnnotations = Collections.emptyList();
+
+    // Classes to process specified by implemented interface
+    private List<String> classesImplementingInterfaces = Collections.emptyList();
+
+    // Classes to process specified by extended superclasses
+    private List<String> classesExtendingClasses = Collections.emptyList();
+
+    // Scans specified JAX-RS Application for classes to process
+    private String classesFromJaxrsApplication;
+
+    // Scans JAX-RS resources for JSON classes to process
+    private boolean classesFromAutomaticJaxrsApplication = false;
+
+    // Allows to speed up classpath scanning by limiting scanning to specified packages
+    private List<String> scanningAcceptedPackages = Collections.emptyList();
+
+    // List of classes excluded from processing
+    private List<String> excludeClasses = Collections.emptyList();
+
+    // Excluded classes specified using glob patterns
+    private List<String> excludeClassPatterns = Collections.emptyList();
+
+    // Only include properties with these annotations
+    private List<String> includePropertyAnnotations = Collections.emptyList();
+
+    // Exclude properties with these annotations
+    private List<String> excludePropertyAnnotations = Collections.emptyList();
+
+    // Library used in JSON classes (e.g., Jackson2, Gson)
+    private JsonLibrary jsonLibrary;
+
+    // Jackson 2 global configuration
+    private Jackson2Configuration jackson2Configuration;
+
+    // Gson global configuration
+    private GsonConfiguration gsonConfiguration;
+
+    // JSON-B global configuration
+    private JsonbConfiguration jsonbConfiguration;
+
+    // Scan Spring REST application for classes to process
+    private boolean scanSpringApplication = false;
+
+    // Specifies the level of logging output (Debug, Verbose, etc.)
+    private Logger.Level loggingLevel = Logger.Level.Verbose;
+
+    // Skip processing
+    private boolean skip = false;
+    private List<String> optionalAnnotations = Collections.emptyList();
```

Comment:
I would love to find a way that we can share the definition of accepted parameters between this and the Maven plugin. Will Gradle allow these configuration parameters to be public? I think if that is the case, these could both extend an abstract class with all of the public properties, commented for Gradle, and annotated for Maven. See the Maven mojo: https://github.com/ivangreene/java-to-zod/blob/main/java-to-zod-maven-plugin/src/main/java/sh/ivan/zod/GenerateZodSchemasMojo.java